Makefile

```diff
@@ -3,17 +3,39 @@
 
 ROOT_DIR=${CURDIR}
 LLVM_PROJ_DIR?=$(ROOT_DIR)/src/llvm-project
+
+# Windows needs munging
+ifeq ($(OS),Windows_NT)
+
+PREFIX?=c:/wasi-sdk
+# we need to explicitly call bash -c for makefile $(shell ...), otherwise we'll try under
+# who knows what
+BASH?=bash -c
+
+ifneq (x$(MSYSTEM),x)
+# msys needs any /-prefixed arguments, or =/ containing, to turn into //
+# because it tries to path-expand the / into the msys root.  // escapes this.
+SLASHY_SLASH=/
+# assuming we're running under msys2 (git-bash), PATH needs /c/foo format directories
+PATH_PREFIX=$(shell cygpath.exe -u $(PREFIX))
+endif
+
+endif
+
 PREFIX?=/opt/wasi-sdk
+PATH_PREFIX?=$(PREFIX)
+SLASHY_SLASH?=
+BASH?=
 
-CLANG_VERSION=$(shell ./llvm_version.sh $(LLVM_PROJ_DIR))
-VERSION:=$(shell ./version.sh)
+CLANG_VERSION=$(shell $(BASH) ./llvm_version.sh $(LLVM_PROJ_DIR))
+VERSION:=$(shell $(BASH) ./version.sh)
 DEBUG_PREFIX_MAP=-fdebug-prefix-map=$(ROOT_DIR)=wasisdk://v$(VERSION)
 
 default: build
 	@echo "Use -fdebug-prefix-map=$(ROOT_DIR)=wasisdk://v$(VERSION)"
 
 check:
-	cd tests && PATH="$(PREFIX)/bin:$$PATH" ./run.sh
+	cd tests && PATH="$(PATH_PREFIX)/bin:$$PATH" ./run.sh
 
 clean:
 	rm -rf build $(PREFIX)
```

Comment:
Should this be PATH_PREFIX too?

Reply:
no, doesn't need to be -- msys understands `c:/foo/bar` just fine as arguments; just in the `PATH` env var, things have to be `/c/foo` because PATH itself is colon-separated